### PR TITLE
Add HTTP.rb adapter

### PR DESCRIPTION
**Pull request by iMacTia**
## Description
This adds an adapter for [HTTP - The Gem! a.k.a http.rb](https://github.com/httprb/http)
Based on the work from @PandaWhisperer in https://github.com/PandaWhisperer/faraday/pull/1

### CODE REVIEW

File: lib/faraday/http/adapter.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+        resp = conn.request env[:method], env[:url], body: env[:body]
+        save_response(env, resp.code, resp.body.to_s, resp.headers, resp.status.reason)
+      rescue ::HTTP::TimeoutError
+        raise Faraday::TimeoutError, $ERROR_INFO
+      rescue ::HTTP::ConnectionError
+        raise Faraday::ConnectionFailed, $ERROR_INFO
+      end
+
+      def setup_connetion(env)
```
Comment (technoweenie) [inline lib/faraday/http/adapter.rb]:
```suggestion
      def setup_connection(env)
```

File: lib/faraday/http/adapter.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        conn = ::HTTP
+
+        request_config(conn, env[:request]) if env[:request]
```
Comment (technoweenie) [inline lib/faraday/http/adapter.rb]: Is this actually configuring the `conn` object? The var gets replaced inside `#request_config`, but is never updated here in `#setup_connetion`.

```suggestion
        conn = request_config(conn, env[:request]) if env[:request]
```

Comment (iMacTia) [reply lib/faraday/http/adapter.rb]: I had to look into the `HTTP.rb` gem, but from what I saw they return a new client every time you call one of these methods 🤦‍♂️
More here: https://github.com/httprb/http/blob/master/lib/http/chainable.rb#L93 (e.g. `timeout`)

File: faraday-http.gemspec [hunk 3]
```diff
@@ -37,8 +37,14 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ['lib']
 
+  # spec.add_dependency 'faraday', '~> 1.0'
```
Comment (olleolleolle) [inline faraday-http.gemspec]:
```suggestion
```

**Comment by iMacTia** [comment]
Just waiting for https://github.com/lostisland/faraday/pull/941 to be merged before merging this in, will need to change the `Gemfile` as well

**Comment by olleolleolle** [comment]
@iMacTia That branch is now merged, this PR is ready for its next step: Gemfile fix.

**Comment by iMacTia** [comment]
Thanks! Will try to have a look today otherwise tomorrow will be done 😁

**Pull request merged**